packages/kit/src/runtime/client/client.js

```diff
@@ -0,0 +1,1249 @@
+				// MEMO: Without this condition, firefox will open mailer twice.
+				// See: https://github.com/sveltejs/kit/issues/4045
+				if (!is_svg_a_element && url.origin === 'null') return;
+
+				// Ignore if tag has
+				// 1. 'download' attribute
+				// 2. 'rel' attribute includes external
+				const rel = (a.getAttribute('rel') || '').split(/\s+/);
+
+				if (a.hasAttribute('download') || (rel && rel.includes('external'))) {
+					return;
+				}
+
+				// Ignore if <a> has a target
+				if (is_svg_a_element ? a.target.baseVal : a.target) return;
+
+				// Check if new url only differs by hash and use the browser default behavior in that case
+				// This will ensure the `hashchange` event is fired
+				// Removing the hash does a full page navigation in the browser, so make sure a hash is present
+				const [base, hash] = url.href.split('#');
+				if (hash !== undefined && base === location.href.split('#')[0]) {
+					// set this flag to distinguish between navigations triggered by
+					// clicking a hash link and those triggered by popstate
+					hash_navigating = true;
+
+					update_scroll_positions(current_history_index);
+					_update_page_store(new URL(url.href));
+
+					return;
+				}
+
+				_navigate({
+					url,
+					scroll: a.hasAttribute('sveltekit:noscroll') ? scroll_state() : null,
+					keepfocus: false,
+					chain: [],
+					details: {
+						state: {},
+						replaceState: false
+					},
+					accepted: () => event.preventDefault(),
+					blocked: () => event.preventDefault()
+				});
+			});
+
+			addEventListener('popstate', (event) => {
+				if (event.state && enabled) {
+					// if a popstate-driven navigation is cancelled, we need to counteract it
+					// with history.go, which means we end up back here, hence this check
+					if (event.state[INDEX_KEY] === current_history_index) return;
+
+					_navigate({
+						url: new URL(location.href),
+						scroll: scroll_positions[event.state[INDEX_KEY]],
+						keepfocus: false,
+						chain: [],
+						details: null,
+						accepted: () => {
+							current_history_index = event.state[INDEX_KEY];
+						},
+						blocked: () => {
+							const delta = current_history_index - event.state[INDEX_KEY];
+							history.go(delta);
+						}
+					});
+				}
+			});
+
+			addEventListener('hashchange', () => {
+				// if the hashchange happened as a result of clicking on a link,
+				// we need to update history, otherwise we have to leave it alone
+				if (hash_navigating) {
+					hash_navigating = false;
+					history.replaceState(
+						{ ...history.state, [INDEX_KEY]: ++current_history_index },
+						'',
+						location.href
+					);
+				}
+			});
+		},
+
+		_hydrate: async ({ status, error, nodes, params }) => {
+			const url = new URL(location.href);
+
+			/** @type {Array<import('./types').BranchNode | undefined>} */
+			const branch = [];
+
+			/** @type {Record<string, any>} */
+			let stuff = {};
+
+			/** @type {import('./types').NavigationResult | undefined} */
```

Comment:
Agree that the terminology could be tightened up a bit, though I think there's value in thinking of a 'load' as something that relates to a single node (i.e. layout/page component) in the branch, since that's where you might find a `load` function